exchange/websocket/manager_test.go

```diff
@@ -1336,3 +1336,18 @@ func TestGetConnection(t *testing.T) {
 	require.NoError(t, err)
 	assert.Same(t, expected, conn)
 }
+
+func TestWebsocketConnectionRequireMatchWithData(t *testing.T) {
+	t.Parallel()
+	ws := connection{Match: NewMatch()}
+	err := ws.RequireMatchWithData(0, nil)
+	require.ErrorIs(t, err, ErrSignatureNotMatched)
+
+	ch, err := ws.Match.Set(0, 1)
+	require.NoError(t, err)
+
+	err = ws.RequireMatchWithData(0, []byte("test"))
+	require.NoError(t, err)
+
+	require.Equal(t, []byte("test"), <-ch)
```

Comment:
🐒 
* I think this is an assert, not a require
* Won't this block badly if there's nothing there?
`d, ok :=`

Reply:
https://github.com/thrasher-corp/gocryptotrader/pull/1613/commits/70cea726c96e36df4bd98f251bed04660cdd112d